app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,95 @@
+class TasksController < ApplicationController
+
+  before_action :find_task, except: [:index, :create]
+
+  def index
+    @tasks = Task.all.order("created_at")
+  end
+
+  def show
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    #instantiate a new task
+    @task = Task.new(task_params)
+
+    # save returns true if the database insert succeeds
+    if @task.save
+      # go to the index so we can see the task in the list, send them back to '/tasks' path
+      redirect_to task_path(@task.id)
+      return
+    else                              # save failed
+      render :new, :bad_request       #show the new task form view again
+      return
+    end
+
+  end
+
+  def edit
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def update
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    elsif @task.update(task_params)
+      redirect_to tasks_path # go to the index so we can see the task in the list
+      return
+    else                # save failed
+      render :edit      # show the new task form view again
+      return
+    end
+  end
+
+  def destroy
+    if @task
+      @task.destroy
+      redirect_to tasks_path
+    else
+      head :not_found
+    end
+  end
+
+  def complete
```

Comment:
Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent.